src/com/videojs/providers/HTTPVideoProvider.as

```diff
@@ -205,16 +205,14 @@ package com.videojs.providers{
             appendBuffer(FLV_HEADER);
         }
 
-        public function get buffered():Number{
-            // _src.path == null when in data generation mode
-            if(_ns && _src.path == null)
-            {
-                return _startOffset + _ns.bufferLength + _ns.time;
-            } else if(duration > 0){
-                return (_ns.bytesLoaded / _ns.bytesTotal) * duration;
```

Comment:
I'm pretty sure backBufferLength is a standard property that is around for all stream types, but I'll double check it. I eliminated the requirement for a duration because as far as I can tell, the previous mechanism for calculating this is incorrect. `bytesLoaded / bytesTotal` assumes you're dealing with a constant bitrate video with uniform video and audio distribution, which isn't universally true. I decided to be a bit more aggressive cleaning up since this would still become part of the 4 to 5 transition.